packages/adapters/wagmi/src/connectors/UniversalConnector.ts

```diff
@@ -276,14 +276,14 @@ export function walletConnect(
       }
 
       try {
-        if (chainToSwitch?.caipNetworkId) {
-          provider.setDefaultChain(chainToSwitch?.caipNetworkId as string)
-        }
-
         await provider.request({
           method: 'wallet_switchEthereumChain',
           params: [{ chainId: numberToHex(chainId) }]
         })
+
+        if (chainToSwitch?.caipNetworkId) {
+          provider.setDefaultChain(chainToSwitch?.caipNetworkId as string)
+        }
```

Comment:
Why move to after?  How does it change things?

Reply:
For some wallets (e.g MM) will emit a `chainChanged` event when this is called and the chainId is not the active chainId. So thats why we first need to switch to the correct chain.